src/Utilities/FlowAnalysis/Analysis/TaintedDataAnalysis/TaintedDataAnalysis.TaintedDataOperationVisitor.cs

```diff
@@ -0,0 +1,472 @@
+            /// <param name="wellKnownTypeProvider">Well known types cache.</param>
+            /// <param name="method">Instance method being called.</param>
+            /// <returns>True if the method returns tainted data, false otherwise.</returns>
+            private bool IsSanitizingMethod(IMethodSymbol method)
+            {
+                foreach (SanitizerInfo sanitizerInfo in this.DataFlowAnalysisContext.SanitizerInfos.GetInfosForType(method.ContainingType))
+                {
+                    if (method.MethodKind == MethodKind.Constructor
+                        && sanitizerInfo.IsConstructorSanitizing)
+                    {
+                        return true;
+                    }
+
+                    if (sanitizerInfo.SanitizingMethods.Contains(method.MetadataName))
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+
+            /// <summary>
+            /// Determines if tainted data passed as arguments to a method enters a tainted data sink.
+            /// </summary>
+            /// <param name="method">Method being invoked.</param>
+            /// <param name="taintedArguments">Arguments passed to the method invocation that are tainted.</param>
+            /// <returns>True if any of the tainted data arguments enters a sink, false otherwise.</returns>
+            private bool IsMethodArgumentASink(IMethodSymbol method, IEnumerable<IArgumentOperation> taintedArguments)
+            {
+                if (method.ContainingType == null || !taintedArguments.Any())
+                {
+                    return false;
+                }
+
+                foreach (SinkInfo sinkInfo in this.DataFlowAnalysisContext.SinkInfos.GetInfosForType(method.ContainingType))
+                {
+                    if (method.MethodKind == MethodKind.Constructor
+                        && sinkInfo.IsAnyStringParameterInConstructorASink
+                        && taintedArguments.Any(a => a.Parameter.Type.SpecialType == SpecialType.System_String))
+                    {
+                        return true;
+                    }
+
+                    if (sinkInfo.SinkMethodParameters.TryGetValue(method.MetadataName, out ImmutableHashSet<string> sinkParameters)
+                        && taintedArguments.Any(a => sinkParameters.Contains(a.Parameter.MetadataName)))
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+
+            /// <summary>
+            /// Determines if a property is a sink.
+            /// </summary>
+            /// <param name="propertyReferenceOperation">Property to check if it's a sink.</param>
+            /// <returns>True if the property is a sink, false otherwise.</returns>
+            private bool IsPropertyASink(IPropertyReferenceOperation propertyReferenceOperation)
```

Comment:
Seems this will benefit if SinkInfos cached all sink properties within it so this method becomes a simple hashset contains operation.